### PR TITLE
[Merged by Bors] - feat: auto-suggest style fixes in CI

### PR DESCRIPTION
**Pull request by alexjbest**
This PR tries to help with the issue of contributors pushing a branch, closing their computer, and then later getting notified about a trivial style issue like, leaving a trailing space or a `set_option pp.proofs true` on somewhere.

With this change, the style linter output is added as github suggestions to your PR, which can be easily applied from the web interface without having to check out the branch again.

These checks run in a separate workflow to the original linters as the suggestions can only be added to PRs, not to branches, so this workflow will not run on branches not yet PRed. This is a slight duplication of work when a branch is part of a PR, but the time is so minimal compared to actually building / tactic linting that we dont think this will have a noticable effect on CI resources. 

The check all files imported suggesters are less likely to trigger if the user simply forgot the file. This is due to a technical limitation of github suggestions; suggestions can only be added to lines modified in the diff. It will at least fix help things if the user alphabetises wrongly (this has actually happened to me before...)
We _could_ have the linters automatically push fixes, but that is more invasive and potentially annoying so we hold off that for now until there is appetite for it.

We also moved the slightly expensive install bibtool step later as that step takes some time compared to the others and its nice to get more instant feedback from the earlier steps if they are going to fail anyway.

---

- [x] depends on: #6568 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jcommelin** [comment]
This looks very useful to me! Could you please make a non-demo version of this PR as well? (And cross-link the two of them.)

**Comment by alexjbest** [comment]
Sure I need to find a couple of minutes to fix the merge conflict. But I was planning to just revert the test commits on this PR seeing as they get squashed anyway, rather than make 2 PRs. Does that sound ok to you or should I make a second one

**Comment by Julian** [comment]
I believe that merge commit fixes the conflicts, hopefully correctly!

**Comment by alexjbest** [comment]
> 

I think there is a small issue in that the bibtool install now occurs twice (I moved it later as the step takes some time compared to the others and its nice to get more instant feedback from the earlier steps)

File: Mathlib/NumberTheory/ADEInequality.lean [hunk 1]
```diff
@@ -194,7 +194,9 @@ theorem lt_three {p q r : ℕ+} (hpq : p ≤ q) (hqr : q ≤ r) (H : 1 < sumInv
 #align ADE_inequality.lt_three ADEInequality.lt_three
 
 theorem lt_four {q r : ℕ+} (hqr : q ≤ r) (H : 1 < sumInv {2, q, r}) : q < 4 := by
-  have h4 : (0 : ℚ) < 4 := by norm_num
+  have h4 : (0 : ℚ) < 4 :=
+  by
```
Comment (alexjbest) [inline Mathlib/NumberTheory/ADEInequality.lean:198]: note that this is not yet covered as the correct fix requires some more work (reflowing lines in a way that keeps them under 100 chars is slightly tricky)

File: Mathlib/NumberTheory/ADEInequality.lean [hunk 2]
```diff
@@ -194,7 +194,9 @@ theorem lt_three {p q r : ℕ+} (hpq : p ≤ q) (hqr : q ≤ r) (H : 1 < sumInv
...
+    norm_num
```
Comment (alexjbest) [inline Mathlib/NumberTheory/ADEInequality.lean:199]:
```suggestion
  have h4 : (0 : ℚ) < 4 := by norm_num
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**